python/django/transit_indicators/views.py

```diff
@@ -52,6 +58,75 @@ def post(self, request, *args, **kwargs):
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
+class OTILanguagesView(APIView):
+    """ Endpoint to GET translated versions of the available languages.
+    """
+
+    renderer_classes = (UnicodeJSONRenderer, BrowsableAPIRenderer)
+
+    def get(self, request, *args, **kwargs):
+        try:
+            languages = dict(LANGUAGES)
+            return Response({'languages': languages, 'default': LANGUAGE_CODE}, status=status.HTTP_200_OK)
+        except Exception as ex:
+            return Response({'error': ex.message}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+# helper function to fetch current timezone locale
+def get_tz_locale():
+    current_language = get_language()
+    # strip country suffix if present
+    found_hyphen = current_language.find('-')
+    if found_hyphen >= 0:
+        current_language = current_language[0:found_hyphen]
+    return Locale(current_language)
+
+class OTITimeZonesView(APIView):
+    """ Endpoing to GET translated names for the available timezones, and
+        POST to set system timezone.
+    """
+
+    renderer_classes = (UnicodeJSONRenderer, BrowsableAPIRenderer)
+
+    def get(self, request, *args, **kwargs):
+        try:
+            # build dictionary of timezone: translated city name
+            locale = get_tz_locale()
+            timezones = [{'zone': tz, 'name': get_timezone_name(tz, locale=locale)}
+                         for tz in pytz.all_timezones]
+            return Response(timezones, status=status.HTTP_200_OK)
+        except Exception as ex:
+            return Response({'error': ex.message}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    def post(self, request, *args, **kwargs):
+        try:
+            timezone = request.DATA.get('timezone')
```

Comment:
Doesn't this throw a `KeyError` if timezone doesn't exist? If so, might be cleaner to handle it that way so that its consistent with all the other thrown exceptions below.